packages/swayze/lib/src/widgets/headers/gestures/header_gesture_detector.dart

```diff
@@ -250,67 +274,253 @@ class _HeaderGestureDetectorState extends State<HeaderGestureDetector> {
     );
   }
 
+  /// Handles drag starts that should start dragging a header around.
+  void handleStartDraggingHeader(
+    DragStartDetails details,
+    Range selectionRange,
+  ) {
+    /// Sets the dragging cursor to be basic.
+    /// Instead of using [SystemMouseCursors.grabbing], we set the basic cursors
+    /// because currently there is no mechanism to globally change the cursor on
+    /// desktop, this means that the cursor would be a closed hand only when
+    /// hovering the header, which would cause a weird change of cursors during
+    /// a drag action, making the user think that something went wrong with the
+    /// action.
+    setCursorState(SystemMouseCursors.basic);
+    Actions.invoke(
+      context,
+      HeaderDragStartIntent(
+        draggingPosition: details.localPosition,
+        headers: selectionRange,
+        axis: widget.axis,
+      ),
+    );
+  }
+
+  /// Handles header dragging updates.
+  void handleUpdateDraggingHeader(
+    DragUpdateDetails gestureDetails,
+    _HeaderGestureDetails details,
+  ) {
+    Actions.invoke(
+      context,
+      HeaderDragUpdateIntent(
+        draggingPosition: gestureDetails.localPosition,
+        header: details.headerPosition,
+        axis: widget.axis,
+      ),
+    );
+  }
+
+  void handleDragEnd(SwayzeHeaderDragState state) {
+    if (state.isDropAllowed) {
+      Actions.invoke(
+        context,
+        HeaderDragEndIntent(
+          header: state.dropAtIndex,
+          axis: widget.axis,
+        ),
+      );
+    } else {
+      handleDragCancel();
```

Comment:
nit: feel free to ignore as this doesn't add much difference, I usually prefer a early exit:

```dart
if (!state.isDropAllowed) {
  return handleDragCancel(); // or have return in the next line
}

Actions.invoke(
   context,
   HeaderDragEndIntent(
      header: state.dropAtIndex,
      axis: widget.axis,
  ), 
);
```